### ISSUE TITLE
Mapping file seems to have no effect.

### ISSUE THREAD

**Issue by Sahnvour** [include-what-you-use/include-what-you-use#480]
I am trying to apply IWYU to a massive codebase, and would like to use a mappings file, but it does not work as I expected.

As I understand, I can use them to change for example C includes to C++ includes as follow :

```json
[
    {include: ["<stddef.h>", "public", "<cstddef>", "public"]},
    {include: ["<string.h>", "public", "<cstring>", "public"]},
    {include: ["<limits.h>", "public", "<climits>", "public"]},
    {include: ["<errno.h>", "public", "<cerrno>", "public"]},
    {include: ["<stdarg.h>", "public", "<cstdarg>", "public"]}
]
```
but I still get recommendations for the C includes as such `#include <stddef.h>                     // for size_t`.

However, if I make symbols mappings like `{symbol: ["size_t", "private", "<cstddef>", "public"]}` then the header used is indeed `<cstddef>`. Do I need to repeat this process for any symbol that may include C headers in order to be sure I get only C++ headers ?
It would be much simpler to use include mappings to "rename" some headers. Not sure I understand fully the goal of this then.

**Comment by kimgr** [comment]
I think if both include-names are marked `public`, they're seen as interchangeable, i.e. IWYU won't replace an existing `<cstddef>` with `<stddef.h>`. But to force replacement the other way around, I think you'll need to say, e.g. 
```json
[
    { "include": ["<stddef.h>", "private", "<cstddef>", "public"]},
    ...
]
```

That, on the other hand, may cause you to run into #414...

**Comment by Sahnvour** [comment]
Yes, exactly. I also tried to pass `--no_default_mappings` without success, I'll give it a try again tomorrow to be sure.

**Comment by Sahnvour** [comment]
Well I must have done something wrong on my last try, this time using `stl.c.headers.imp` with a `private` initial visibility and `--no_default_mappings` does the trick. So what options am I left with, disabling default mappings and copying all the builtins (plus minor modifications where needed) locally ?

**Comment by kimgr** [comment]
Good to hear you got it to work. Yes, I think so. I'd like to relax the behavior so that external mapping files can overwrite built-in mappings (the current assertion should only apply for conflicting built-in mappings), but I haven't got around to it yet.
